frontend/packages/console-app/src/components/dashboards-page/console-plugins-health-resource/status.ts

```diff
@@ -0,0 +1,19 @@
+import { HealthState } from '@console/shared/src/components/dashboard/status-card/states';
+import { pluginStore } from '../../../../../../public/plugins';
+
+export const getDynamicPluginHealthState = (): { state: HealthState } => {
+  const dynamicPluginInfo = pluginStore.getDynamicPluginInfo();
+  if (dynamicPluginInfo.some((plugin) => plugin.status === 'Failed')) {
+    return { state: HealthState.ERROR };
```

Comment:
I wonder if we want this to be:

- If <100% fail, warning
- If 100% fail, error

If some of the plugins are still working, this should be different than all of them have failed.

Reply:
> Running locally looks weird lol -- I am on the fence if we need to fix this as it's a development oddity. I'm worried we'll get bug reports from others running it locally. ![Screen Shot 2022-06-10 at 2 02 08 PM](https://user-images.githubusercontent.com/8126518/173125346-984f6a67-4445-4c09-b068-7260fb3509b9.png)
> 
> If I `oc apply` the `oc-manifiest` for the Dynamic Demo Plugin... it gets even more weird.
> 
> ![Screen Shot 2022-06-10 at 2 12 28 PM](https://user-images.githubusercontent.com/8126518/173126089-fcfc404c-8090-4ecf-9126-63fe0a0fe677.png) ![Screen Shot 2022-06-10 at 2 12 20 PM](https://user-images.githubusercontent.com/8126518/173126104-ba7601f5-4214-41c8-bc5f-492545b3c2ca.png)
> 
> Again, a local running oddity, but it's starting to showcase the data is not accurately representative. I wonder if this is a way to determine the state of things. Maybe looking at the operator config & comparing that to the number of ConsolePlugins CRs (and their names).

I noticed that as well but chalked it up to development due to the way plugins run for development purposes.  I'm certainly not opposed to looking for a fix, but assumed it wouldn't happen in production so didn't prioritize fixing it.

> Did we not want inactive plugins as well? I thought that was a desired thing as well from the screenshots. ![image](https://user-images.githubusercontent.com/8126518/173125502-b0e92b50-04aa-4d87-aa5e-8f2322d6cc4d.png)

Looking at this again with fresh eyes, I think we actually want up to three categories in the popover that represent the `status` values: `Failed`, `Pending`, `Loaded`.  So maybe something like:

```
Plugin status
A brief description about dynamic plugins.

Failed plugins
* plugin-name

Pending plugins
* plugin-name-2

Loaded plugins                             1/1 enabled
[View all](http://localhost:9000/k8s/cluster/operator.openshift.io~v1~Console/cluster/console-plugins)
```

where `Failed` and `Pending` categories only appear if one or more plugins match that status.  Without these optional categories, users will never be able to know which plugins are causing the status of the dashboard item itself.

Reply:
> I wonder if we want this to be:
> 
> * If <100% fail, warning
> * If 100% fail, error
> 
> If some of the plugins are still working, this should be different than all of them have failed.

I believe the behavior as I've coded it is consistent with the other status items as the `error` description below the item is `Degraded`.

<img width="425" alt="Screen Shot 2022-06-09 at 2 28 53 PM" src="https://user-images.githubusercontent.com/895728/173589811-210752e1-670b-4bce-8f01-81f2ea4da016.png">

Reply:
We need to show what plugin is "degraded" for that flow to work... but I'm not against the design direction